### PR TITLE
feat(client): export HTTP request metrics and logs

### PR DESCRIPTION
**Pull request by giac-mysten**
Contributes to #496

### CODE REVIEW

File: crates/walrus-service/bin/client.rs [hunk 1]
```diff
@@ -309,9 +310,13 @@ struct RpcArg {
 #[serde(rename_all = "camelCase")]
 struct DaemonArgs {
     /// The address to which to bind the service.
-    #[clap(short, long, default_value_t = default::bind_address())]
+    #[clap(long, default_value_t = default::bind_address())]
```
Comment (mlegner) [inline crates/walrus-service/bin/client.rs]: Important: In the docs we currently use the short `-b` in a few places. These need to be updated if we remove the short option.

Comment (giac-mysten) [reply crates/walrus-service/bin/client.rs]: thanks for the catch @mlegner . Let me see what the best option is.

Comment (giac-mysten) [reply crates/walrus-service/bin/client.rs]: I changed the `short` for `--metrics-address` to `-a`, and reintroduced `-b` for `--bind-address`.

File: crates/walrus-service/bin/client.rs [hunk 2]
```diff
@@ -467,6 +478,19 @@ fn init_tracing_subscriber() -> Result<()> {
     Ok(())
 }
 
+// Starts the Prometheus server and returns the registry.
+fn run_registry_and_telemetry(mut metrics_address: SocketAddr) -> Registry {
+    metrics_address.set_ip(IpAddr::V4(Ipv4Addr::UNSPECIFIED));
+    let registry_service = mysten_metrics::start_prometheus_server(metrics_address);
+    let registry = registry_service.default_registry();
+    let (_telemetry_guards, _tracing_handle) = telemetry_subscribers::TelemetryConfig::new()
```
Comment (jpcsmith) [inline crates/walrus-service/bin/client.rs]: Important: The guards cannot be dropped, they must be kept alive.

Comment (giac-mysten) [reply crates/walrus-service/bin/client.rs]: whops. did not consider that, sorry. should be fixed here: 4f8ed461fb2b34549b276a0c49b165bfc3e2d402

**Pull request merged**